libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/map/widgets/logo/CarLogoSurfaceRenderer.kt

```diff
@@ -1,29 +1,33 @@
 package com.mapbox.androidauto.car.map.widgets.logo
 
-import com.mapbox.androidauto.car.internal.extensions.getStyle
-import com.mapbox.androidauto.car.internal.extensions.getStyleAsync
-import com.mapbox.maps.LayerPosition
+import android.graphics.Rect
+import com.mapbox.maps.EdgeInsets
 import com.mapbox.maps.MapboxExperimental
 import com.mapbox.maps.extension.androidauto.MapboxCarMapObserver
 import com.mapbox.maps.extension.androidauto.MapboxCarMapSurface
+import com.mapbox.maps.extension.androidauto.widgets.LogoWidget
+import com.mapbox.maps.renderer.widget.WidgetPosition
 
 @OptIn(MapboxExperimental::class)
-class CarLogoSurfaceRenderer(
-    private val layerPosition: LayerPosition? = null
-) : MapboxCarMapObserver {
+class CarLogoSurfaceRenderer : MapboxCarMapObserver {
+
+    private var logoWidget: LogoWidget? = null
 
     override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
-        val logoWidget = LogoWidget(mapboxCarMapSurface.carContext)
-        mapboxCarMapSurface.getStyleAsync {
-            it.addPersistentStyleCustomLayer(
-                LogoWidget.LOGO_WIDGET_LAYER_ID,
-                logoWidget.host,
-                layerPosition
-            )
-        }
+        val logoWidget = LogoWidget(
+            mapboxCarMapSurface.carContext,
+            WidgetPosition(WidgetPosition.Horizontal.RIGHT, WidgetPosition.Vertical.BOTTOM),
+            marginX = 26f, marginY = 10f,
```

Comment:
Thinking about cleaning this up a little, considering this will be a more permanent class. What do you think about adding the logo widget to the constructor, and adding the default to a companion object?

We could remove `private var logoWidget: LogoWidget? = null` and make a default in the constructor.

``` kotlin
class CarLogoSurfaceRenderer(
  val logoWidget: LogoWidget = defaultLogoWidget
) : MapboxCarMapObserver {

  companion object {
    val defaultLogoWidget = LogoWidget(...
  }
}
```

Reply:
Oh, it is constructed when `onAttached` is called so that idea is not great.

Smaller change, would be to move the constants to a companion object so the magic numbers are a little more visible. These margin values are going to need to change in order to support different screen resolutions

``` kotlin
class CarLogoSurfaceRenderer(
) : MapboxCarMapObserver {

  private companion object {
    private val defaultWidgetPosition = WidgetPosition(...
    private const val defaultMarginX = 26f
    private const val defaultMarginY = 10f
  }
}
```

Reply:
Not sure about the naming, `default` suggest that these values can be customized, but they aren't. I think we can get back to it, when we add support for more screen resolutions.

Reply:
Not sure about the naming, `default` suggests that these values can be customized, however they aren't. I think we can get back to it, when we add support for more screen resolutions.